pyomo/contrib/mindtpy/feasibility_pump.py

```diff
@@ -213,20 +200,21 @@ def feas_pump_loop(solve_data, config):
         else:
             raise ValueError(
                 'MindtPy unable to handle NLP subproblem termination '
-                'condition of {}'.format(termination_condition))
+                'condition of {}'.format(fp_nlp_result.solver.termination_condition))
         # Call the NLP post-solve callback
         config.call_after_subproblem_solve(fp_nlp, solve_data)
         solve_data.fp_iter += 1
-    solve_data.mip.MindtPy_utils.feas_pump_mip_obj.deactivate()
-    
+    solve_data.mip.MindtPy_utils.del_component('feas_pump_mip_obj')
+
     if config.fp_master_norm == 'L1':
         solve_data.mip.MindtPy_utils.del_component("L1_objective_function")
     elif config.fp_master_norm == 'L_infinity':
-        solve_data.mip.MindtPy_utils.del_component("L_infinity_objective_function")
-    
+        solve_data.mip.MindtPy_utils.del_component(
+            "L_infinity_objective_function")
+
     # deactivate the improving_objective_cut
-    if solve_data.mip.MindtPy_utils.MindtPy_linear_cuts.find_component('improving_objective_cut') is not None:
-        solve_data.mip.MindtPy_utils.MindtPy_linear_cuts.improving_objective_cut.deactivate()
+    solve_data.mip.MindtPy_utils.MindtPy_linear_cuts.del_component(
+        'improving_objective_cut')
```

Comment:
Here improving_objective_cut is a constraint. In Pyomo, if a constraint is defined, it's hard for us to change it. (I didn't find any good example of this). A more applicable way is to redefine it. However, if we redefine it without deleting it in advance, a warning will show up.

```
WARNING: Implicitly replacing the Component attribute c (type=<class
    'pyomo.core.base.constraint.SimpleConstraint'>) on block unknown with a
    new Component (type=<class
    'pyomo.core.base.constraint.AbstractSimpleConstraint'>). This is usually
    indicative of a modelling error. To avoid this warning, use
    block.del_component() and block.add_component().
```
However, for `Objective` in Pyomo, we can change its expression after it is defined.

Reply:
I understand, alright. Let's keep going this way.